### DISCUSSION TITLE
Breadcrumbs

### DISCUSSION THREAD

**Discussion by bameyrick** [ONSdigital/design-system#43] [Feedback]
Use this place to discuss the [breadcrumbs component](https://ons-design-system.netlify.com/components/breadcrumb) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by jrbarnes9** [comment]
Deep levels of breadcrumbs take up too much space on mobile:
<img width="349" alt="Screenshot 2019-05-14 at 14 32 54" src="https://user-images.githubusercontent.com/43346934/57702101-3da48280-7655-11e9-9693-5ea7729d7f2d.png">

A solution e.g. to display only the parent page link is required for under 500px breakpoint e.g...
<img width="359" alt="Screenshot 2019-05-14 at 14 36 42" src="https://user-images.githubusercontent.com/43346934/57702322-bc99bb00-7655-11e9-8ee0-17a0a17af367.png">

**Comment by bameyrick** [comment]
This layout issue wouldn't be unique to mobile, as this wrapping will happen on all viewports on pages with deep levels of nesting and/or with very long page titles, so I don't think changing to only a parent page link on only breadcrumbs solves the larger problem that deep nesting and long page names make this component take up more space.

**Comment by bameyrick** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [ Deep levels of breadcrumbs take up too much space on mobile:](https://trello.com/c/FasLREFx/277-deep-levels-of-breadcrumbs-take-up-too-much-space-on-mobile)

**Comment by jrbarnes9** [comment]
Breadcrumbs chevrons hidden when screen colours inverted (using v12 ZoomFusion)

**Comment by jrbarnes9** [comment]
Breadcrumbs not rendering correctly in IE11: 
<img width="241" alt="Screenshot 2019-08-27 at 14 08 51" src="https://user-images.githubusercontent.com/43346934/63781825-48bcaf00-c8e2-11e9-98df-a75df9654be1.png">

**Comment by trevorsaint** [comment]
Breadcrumbs not rendering correctly at breakpoint 500px and below.

![image](https://user-images.githubusercontent.com/4989027/83241813-4948e580-a193-11ea-9168-9e817932045b.png)

**Comment by trevorsaint** [comment]
We should remove the 'current page' from the breadcrumb. It adds no real value other than visual distraction.

**Problem**
![image](https://user-images.githubusercontent.com/4989027/84881773-3cc6f700-b086-11ea-8b86-0b33cf06711b.png)

**Improvement**
![image](https://user-images.githubusercontent.com/4989027/84881877-6122d380-b086-11ea-87d5-06fd82ef0915.png)

GDS have removed the current page.
https://design-system.service.gov.uk/components/breadcrumbs/

**Comment by jrbarnes9** [comment]
Consider aligning breadcrumbs with previous link used in questionnaires, and consider use of back links in transactions

**Comment by trevorsaint** [comment]
We need to remove ID’s from SVG’s unless absolutely required, which don’t appear to be. 

This is causing W3C validation errors due to duplicate ID’s.

**Comment by trevorsaint** [comment]
Breadcrumb arrow does not align with the edge of the page content when in a single breadcrumb state.

A negative left margin of -6px will perfectly align it.

![image](https://user-images.githubusercontent.com/4989027/93458049-b5349280-f8d7-11ea-83d7-0da65b3d2a6f.png)

**Comment by trevorsaint** [comment]
I am not sure why we have ID’s on the breadcrumbs. But having empty IDs on the links will cause the page to break W3C validation.

<img width="816" alt="Screenshot 2020-12-10 at 16 32 24" src="https://user-images.githubusercontent.com/4989027/101800686-6fb5e900-3b05-11eb-97de-969f51ac6e08.png">

**Comment by jrbarnes9** [comment]
Need to be able to add classes to breadcrumb links
